packages/events/defaultSubscribers/dashboard.js

```diff
@@ -41,9 +43,7 @@ module.exports = {
     ],
     "rpc:request": [
       function (event) {
-        if (!isDashboardNetwork(this.config)) {
-          return;
-        }
+        if (!isDashboardNetwork(this.config)) return;
```

Comment:
Do you think this is the right place for this, or should it be closer to where the `rpc:request` is emitted? If filtering can be done close to the source, then we don't have to expend an event.

Reply:
I believe it belongs here. Wherever `rpc:request` (or any other) event is emitted, the primary concern is that the action of emitting is carried out, not what subscribes to the events and what they decide to do with it.

More importantly I think it complicates the logic at source. E.g. for `rpc:request`, checking that the network isn't dashboard doesn't mean the event shouldn't be emitted, it just means dashboard should ignore it. It should still be emitted because there are other workflows that depend on this event and have event handlers setup for it.

Reply:
On a side note I do think repeatedly writing `if (!isDashboardNetwork(this.config)) return;` isn't ideal. Maybe when dashboard has more event handlers that need such setup we can extract this logic into a decorator / wrapper.